qovery shell --command panics when no TTY is available

## Problem

When `--command` is provided to `qovery shell`, the CLI panics immediately if stdin is not a TTY:

```
panic: provided file is not a console
goroutine 1 [running]:
github.com/containerd/console.Current()
    .../console.go:80 +0x94
github.com/qovery/qovery-cli/pkg.ExecShell(...)
    .../pkg/shell.go:67 +0x16e
```

## Context

We're building an internal developer tooling workflow where `qovery shell --command` is used to exec a command into a running container non-interactively — specifically, to launch an AI coding agent (Claude Code CLI) inside our remote dev environments without a human-in-the-loop terminal session.

## Root Cause

`pkg/shell.go` calls `console.Current()` unconditionally, which requires a real TTY. When `--command` is explicitly provided, a PTY is not needed — the command just needs stdin/stdout piped through, exactly like `kubectl exec -i` (without `-t`). PTY allocation only makes sense for interactive sessions (no `--command`).

## Proposed Fix

Gate the `console.Current()` call on whether the session is interactive. If `--command` is provided, skip PTY allocation and use plain stdin/stdout piping instead.

This would unlock legitimate automation use cases — CI pipelines, agent-driven workflows, scripted provisioning — where `qovery shell --command` is the natural interface but a TTY is simply not available.

Happy to contribute a PR if that would help.

--- a/go.mod
+++ b/go.mod
@@ -32,6 +32,7 @@ require (
 	github.com/tonistiigi/go-rosetta v0.0.0-20220804170347-3f4430f2d346
 	github.com/xlab/treeprint v1.2.0
 	golang.org/x/sys v0.40.0
+	golang.org/x/term v0.39.0
 	gopkg.in/yaml.v3 v3.0.1
 	k8s.io/apimachinery v0.35.0
 	k8s.io/client-go v0.35.0
@@ -99,7 +100,6 @@ require (
 	go4.org v0.0.0-20260112195520-a5071408f32f // indirect
 	golang.org/x/net v0.49.0 // indirect
 	golang.org/x/oauth2 v0.34.0 // indirect
-	golang.org/x/term v0.39.0 // indirect
 	golang.org/x/text v0.33.0 // indirect
 	golang.org/x/time v0.14.0 // indirect
 	google.golang.org/protobuf v1.36.11 // indirect

--- a/pkg/shell.go
+++ b/pkg/shell.go
@@ -4,6 +4,7 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"io"
 	"net/http"
 	"net/url"
 	"os"
@@ -19,6 +20,7 @@ import (
 	"github.com/gorilla/websocket"
 	"github.com/qovery/qovery-cli/utils"
 	log "github.com/sirupsen/logrus"
+	"golang.org/x/term"
 )
 
 const StdinBufferSize = 4096
@@ -67,24 +69,40 @@ func ExecShell(req TerminalSize, path string) {
 		cancel()
 	}()
 
-	currentConsole := console.Current()
-	defer func() {
-		_ = currentConsole.Reset()
-	}()
+	// Allocate a PTY only when stdin is a real terminal. When piped (e.g.
+	// `qovery shell --command ... < input` or invoked from automation),
+	// containerd/console.Current() panics with "provided file is not a console".
+	// In that case we behave like `kubectl exec -i`: pipe os.Stdin/os.Stdout
+	// straight through and leave TtyWidth/TtyHeight at zero so the server
+	// does not attempt to allocate a TTY on the remote side either.
+	interactive := term.IsTerminal(int(os.Stdin.Fd()))
 
-	if err := currentConsole.SetRaw(); err != nil {
-		log.Fatal("error while setting up console", err)
-	}
+	var stdinReader io.Reader = os.Stdin
+	var stdoutWriter io.Writer = os.Stdout
 
-	winSize, err := currentConsole.Size()
-	if err != nil {
-		log.Fatal("Cannot get terminal size", err)
+	if interactive {
+		currentConsole := console.Current()
+		defer func() {
+			_ = currentConsole.Reset()
+		}()
+
+		if err := currentConsole.SetRaw(); err != nil {
+			log.Fatal("error while setting up console", err)
+		}
+
+		winSize, err := currentConsole.Size()
+		if err != nil {
+			log.Fatal("Cannot get terminal size", err)
+		}
+		req.SetTtySize(winSize.Width, winSize.Height)
+
+		stdinReader = currentConsole
+		stdoutWriter = currentConsole
 	}
-	req.SetTtySize(winSize.Width, winSize.Height)
 
 	stdIn := make(chan []byte)
 	wg.Add(1)
-	go readUserConsole(ctx, cancel, currentConsole, stdIn, &normalExit, &wg)
+	go readUserConsole(ctx, cancel, stdinReader, interactive, stdIn, &normalExit, &wg)
 
 	for {
 		if ctx.Err() != nil || userCancelled.Load() || normalExit.Load() {
@@ -107,7 +125,7 @@ func ExecShell(req TerminalSize, path string) {
 
 		done := make(chan struct{})
 		wg.Add(1)
-		go readWebsocketConnection(ctx, cancel, wsConn, currentConsole, done, &normalExit, &wg)
+		go readWebsocketConnection(ctx, cancel, wsConn, stdoutWriter, done, &normalExit, &wg)
 
 		pingTicker := time.NewTicker(PingInterval)
 
@@ -178,7 +196,7 @@ func createWebsocketConn(req interface{}, path string) (*websocket.Conn, error) 
 	return conn, err
 }
 
-func readWebsocketConnection(ctx context.Context, cancel context.CancelFunc, wsConn *websocket.Conn, currentConsole console.Console, done chan struct{}, normalExit *atomic.Bool, wg *sync.WaitGroup) {
+func readWebsocketConnection(ctx context.Context, cancel context.CancelFunc, wsConn *websocket.Conn, out io.Writer, done chan struct{}, normalExit *atomic.Bool, wg *sync.WaitGroup) {
 	defer wg.Done()
 
 	var once sync.Once
@@ -242,7 +260,7 @@ func readWebsocketConnection(ctx context.Context, cancel context.CancelFunc, wsC
 				continue
 			}
 
-			if _, err = currentConsole.Write(msg); err != nil {
+			if _, err = out.Write(msg); err != nil {
 				log.Errorf("error while writing in console: %v", err)
 				return
 			}
@@ -250,7 +268,7 @@ func readWebsocketConnection(ctx context.Context, cancel context.CancelFunc, wsC
 	}
 }
 
-func readUserConsole(ctx context.Context, cancel context.CancelFunc, currentConsole console.Console, stdIn chan []byte, normalExit *atomic.Bool, wg *sync.WaitGroup) {
+func readUserConsole(ctx context.Context, cancel context.CancelFunc, in io.Reader, interactive bool, stdIn chan []byte, normalExit *atomic.Bool, wg *sync.WaitGroup) {
 	defer wg.Done()
 
 	buffer := make([]byte, StdinBufferSize)
@@ -262,8 +280,14 @@ func readUserConsole(ctx context.Context, cancel context.CancelFunc, currentCons
 			return
 		}
 
-		count, err := currentConsole.Read(buffer)
+		count, err := in.Read(buffer)
 		if err != nil {
+			// In non-interactive mode (piped stdin), EOF just means the input
+			// stream is exhausted; the remote command may still produce output,
+			// so let the websocket loop run until the server closes the session.
+			if !interactive && errors.Is(err, io.EOF) {
+				return
+			}
 			log.Error("error while reading on console:", err)
 			cancel()
 			return

--- a/pkg/shell.go
+++ b/pkg/shell.go
@@ -282,10 +282,25 @@ func readUserConsole(ctx context.Context, cancel context.CancelFunc, in io.Reade
 
 		count, err := in.Read(buffer)
 		if err != nil {
-			// In non-interactive mode (piped stdin), EOF just means the input
-			// stream is exhausted; the remote command may still produce output,
-			// so let the websocket loop run until the server closes the session.
+			// In non-interactive mode (piped stdin), EOF means the input stream
+			// is exhausted but the remote command may still produce output. We
+			// flush any buffered bytes, then send EOT (0x04) so the remote PTY
+			// line discipline propagates EOF to the remote process. Without
+			// this, commands like `cat < file` or `sh -s < script.sh` hang
+			// forever. The websocket loop stays open so we can drain remote
+			// stdout until the server closes the session.
 			if !interactive && errors.Is(err, io.EOF) {
+				if len(pendingBytes) > 0 {
+					select {
+					case <-ctx.Done():
+						return
+					case stdIn <- pendingBytes:
+					}
+				}
+				select {
+				case <-ctx.Done():
+				case stdIn <- []byte{0x04}:
+				}
 				return
 			}
 			log.Error("error while reading on console:", err)

